### PR TITLE
Lt 5621 recreate snapshot

### PR DESCRIPTION
**Pull request by gponomarev-lykke**

### CODE REVIEW

File: src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        public async Task SetShouldRecreateSnapshot(bool value)
+        {
+            await _retryPolicy.ExecuteAsync(() => _database
+                .StringSetAsync(RedisKey, value.ToString(), when: When.Always)); 
+        }
+        
+        public async Task<bool> GetShouldRecreateSnapshot()
+        {
+            var value = await _retryPolicy.ExecuteAsync(() =>
```
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs:39]: How is it gonna work if there is no such a key in Redis?

File: src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public async Task SetShouldRecreateSnapshot(bool value)
+        {
+            await _retryPolicy.ExecuteAsync(() => _database
+                .StringSetAsync(RedisKey, value.ToString(), when: When.Always)); 
+        }
+        
+        public async Task<bool> GetShouldRecreateSnapshot()
+        {
+            var value = await _retryPolicy.ExecuteAsync(() => 
+              _database.StringGetAsync(RedisKey));
```
Comment (tarurar) [inline src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs:40]: Is it going to be frequent operation? Does it make sense to keep the copy of the value in `SnapshotTrackerService` in nullable private field?

File: src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs (hunk 1 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs:39]: As I understand it, a special RedisValue.Null value will be returned, and the subsequent check against bool.TrueString will return false
So, the system will behave like snapshot recreation is not needed.

File: src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs (hunk 2 above)
Comment (gponomarev-lykke) [reply src/MarginTrading.Backend.Services/Services/SnapshotTrackerService.cs:40]: Setting the value will be frequent, but getting it should happen only once during EOD. I think caching the value would be a bit more dangerous in this case, we don't want stale data

**Review by tarurar — APPROVED** [review]

**Pull request merged**